packages/application/src/layoutrestorer.ts

```diff
@@ -52,9 +54,9 @@ export interface ILayoutRestorer {
    *
    * @param options - The restoration options.
    */
-  restore(
-    tracker: InstanceTracker<any>,
-    options: ILayoutRestorer.IRestoreOptions<any>
+  restore<T extends Widget = Widget>(
```

Comment:
I *almost* want the layout restorer to understand non-widgets so it can be used for other kinds of state. But then it would be much more than a layout restorer, and probably too big of a change than we want to make right now.

Reply:
I see you have gone ahead and done this. I'm looking forward to using it!